specs/GetFavicon.md

```diff
@@ -0,0 +1,172 @@
+
+# Description
+We propose a new Webview2 event which would allow developers to access the current Favicon of a page and be notified when the favicon changes. This means when a page first loads, it would raise the FaviconChanged event since before parsing the HTML document there is no known favicon. DOM or JavaScript may change the Favicon, causing the event to be raised again for the same document.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+```cpp
+Gdiplus::GdiplusStartupInput gdiplusStartupInput;
+
+// Initialize GDI+.
+Gdiplus::GdiplusStartup(&gdiplusToken_, &gdiplusStartupInput, NULL);
+CHECK_FAILURE(m_webView2->add_FaviconChanged(
+    Callback<ICoreWebView2ExperimentalFaviconChangedEventHandler>(
+        [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+            if (m_faviconChanged)
+            {
+                wil::unique_cotaskmem_string url;
+                Microsoft::WRL::ComPtr<ICoreWebView2>
+                    webview2Experimental;
+                CHECK_FAILURE(
+                    sender->QueryInterface(IID_PPV_ARGS(&webview2)));
+
+                CHECK_FAILURE(webview2Experimental->get_FaviconUri(&url));
+                std::wstring strUrl(url.get());
+
+                webview2->GetFavicon(
+                    COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG,
+                    Callback<ICoreWebView2ExperimentalGetFaviconCompletedHandler>(
+                        [this, strUrl](HRESULT errorCode, IStream* iconStream) -> HRESULT
+                        {
+                            CHECK_FAILURE(errorCode);
+                            Gdiplus::Bitmap iconBitmap(iconStream);
+                            wil::unique_hicon icon;
+                            if (!iconBitmap.GetHICON(&icon))
+                            {
+                                m_favicon = std::move(icon);
+                                SendMessage(
+                                    m_appWindow->GetMainWindow(), WM_SETICON,
+                                    ICON_SMALL, (LPARAM)m_favicon.get());
+                                m_statusBar.Show(strUrl);
+                            }
+
+                            return S_OK;
+                        })
+                        .Get());
+            }
+            return S_OK;
+        }).Get(), &m_faviconChangedToken));
+}
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    IRandomAccessStream stream = await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
```

Comment:
C# sample should show equivalent functionality to C++ sample. Otherwise people may assume that "Oh, that's not possible in C#. You have to use C++."

In this case, you need to update the status bar with the FaviconUrl.